remotespark/kernels/wrapperkernel/usercodeparser.py

```diff
@@ -5,36 +5,32 @@
 
 
 class UserCodeParser(object):
-    @staticmethod
-    def get_code_to_run(code):
-        # We assume that every single cell will be either a cell magic or a Spark code cell.
-        # Because cell magic bodies need to have *some* content, we append a '\n' at the end.
-        lines = code.split("\n")
-        all_but_first = "\n".join(lines[1:])
+    # A list of the names of all magics that are cell magics, but which have no cell body input.
+    # For example, the %%info magic has no cell body input, i.e. it is incorrect to call
+    #    %%info
+    #    some_input
+    _magics_with_no_cell_body = [i.__name__ for i in [KernelMagics.info, KernelMagics.logs, KernelMagics.cleanup,
+                                                      KernelMagics.delete, KernelMagics.help]]
 
-        if code.startswith("%%local"):
-            return "{}\n ".format(all_but_first)
-        elif code.startswith("%local"):
-            return "{}\n ".format(all_but_first)
-        elif code.startswith("%%"):
+    def get_code_to_run(self, code):
+        try:
+            all_but_first_line = code.split(None, 1)[1]
+        except IndexError:
+            all_but_first_line = ""
+
+        if code.startswith("%%local") or code.startswith("%local"):
+            return all_but_first_line
+        elif any(code.startswith("%%" + s) for s in self._magics_with_no_cell_body):
             return "{}\n ".format(code)
-        elif code.startswith(UserCodeParser.get_kernel_magic_names_as_line_magics()):
```

Comment:
How are you doing this filter now?

```
# We could choose to transform all line magics to cell magics here since we expect that there's no line		
# magics to call, except that there are... for example %autosave and etcetera.		
# So, we'll just transform our magics.
```